### PR TITLE
timer: support cron syntax for timer framework

### LINKED ISSUE #46330: Support cron syntax for timer framework

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #46330

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: timer/api/timer.go [hunk 1]
```diff
@@ -211,3 +272,13 @@ func (r *TimerRecord) Clone() *TimerRecord {
 	cloned.TimerSpec = *r.TimerSpec.Clone()
 	return &cloned
 }
+
+// ValidateTimeZone validates the TimeZone field.
+func ValidateTimeZone(tz string) error {
+	if tz != "" && !strings.EqualFold(tz, TimeZoneTiDB) {
+		if _, ok := timeutil.ParseTimeZone(tz); !ok {
+			return errors.Errorf("Invalid timezone '%s'", tz)
```
Comment (bb7133) [inline timer/api/timer.go:280]: IMO it's better to use `ErrUnknownTimeZone`? `errors.Errorf` gives 'unknown error code' which we should always avoid when possible.

And, is there any specific reason we need to change the return value of `ParseTimeZone` from `error` to `bool`?

File: timer/api/timer.go [hunk 2]
```diff
@@ -15,18 +15,26 @@
...
+	"strings"
 	"time"
 
 	"github.com/pingcap/errors"
 	"github.com/pingcap/tidb/parser/duration"
+	"github.com/pingcap/tidb/util/timeutil"
+	"github.com/robfig/cron/v3"
 )
 
+// TimeZoneTiDB indicates to use the tidb timezone.
```
Comment (bb7133) [inline timer/api/timer.go:27]: It seems that `TimeZoneTiDB` is the same with an empty string for 'timezone' fields in timer. IMO its semantic is not better than an empty string to me -_-.

How about just removing it and explicitly explain that in the client api? For example we can declare that 'if timezone is empty string, we use the (global) `time_zone` system variable'

Comment (lcwangchao) [reply timer/api/timer.go:27]: In the previous implement of timer, it uses `TiDB` as the default value and has already persisted into `mysql.tidb_timers`. 

But I think we can just support parsing `TiDB` value without  supporting write `TiDB` for compatitability

File: timer/api/timer.go (hunk 1 above)
Comment (lcwangchao) [reply timer/api/timer.go:280]: `ParseTimeZone` returns bool is easy to write without introducing another `errors.go` file... But anyway it not very hard to do it, I'll update the PR.

**Review by bb7133 — APPROVED** [review]
LGTM

**Review by YangKeao — APPROVED** [review]
LGTM

**Comment by easonn7** [comment]
/approve

there is no sysvar behavior change.

**Pull request merged**